integration_tests/test_validations.py

```diff
@@ -0,0 +1,80 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import shutil
+
+import fixtures
+
+import integration_tests
+
+
+class ValidationsTestCase(integration_tests.StoreTestCase):
+
+    def setUp(self):
+        super().setUp()
+        keys_dir = os.path.join(os.path.dirname(__file__), 'keys')
+        temp_keys_dir = os.path.join(self.path, '.snap', 'gnupg')
+        shutil.copytree(keys_dir, temp_keys_dir)
+        self.useFixture(fixtures.EnvironmentVariable(
+            'SNAP_GNUPG_HOME', temp_keys_dir))
+
+    def test_gated_success(self):
+        self.addCleanup(self.logout)
+        self.login()
+        self.assertEqual(0, self.gated('ubuntu-core', [
+            ('snap-1', '3'),
+            ('snap-2', '5'),
+        ]))
+
+    def test_gated_no_login_failure(self):
+        self.assertEqual(1, self.gated('ubuntu-core', [
+            ('snap-1', '3'),
+            ('snap-2', '5'),
+        ], expected_error='Have you run "snapcraft login'))
```

Comment:
Like this call, the expected_validations is not relevant here, only the expected_error